python/docs/cli.rst

```diff
@@ -0,0 +1,231 @@
+
+
+Branch Command
+--------------
+Perform operations on branches: create, delete, modify and reassign.
+
+.. code-block:: bash
+
+    Usage: nessie branch [OPTIONS] [BRANCH] [NEW_BRANCH]
+
+      Branch operations.
+
+      BRANCH name of branch to list or create/assign NEW_BRANCH name of branch
+      to assign from or rename to
+
+      Examples:
+
+          nessie branch -l -> list all branches
+
+          nessie branch -l main -> list only main
+
+          nessie branch -d main -> delete main
+
+          nessie branch -> list all branches
+
+          nessie branch main -> create branch main at current head
+
+          nessie branch main test -> create branch main at head of test
+
+          nessie branch -f main test -> assign main to head of test
+
+    Options:
+      -l, --list     list branches
+      -d, --delete   delete a branch
+      -f, --force    force branch assignment
+      --json         write output in json format.
+      -v, --verbose  Verbose output.
+      -c, --condition TEXT  Conditional Hash. Only perform the action if branch
+                            currently points to condition.
+      --help         Show this message and exit.
+
+
+Tag Command
+-----------
+Perform operations on tags: create, delete, modify and reassign.
+
+.. code-block:: bash
+
+    Usage: cli nessie [OPTIONS] [TAG_NAME] [NEW_TAG]
+
+      Tag operations.
+
+      TAG_NAME name of branch to list or create/assign
+
+      NEW_TAG name of branch to assign from or rename to
+
+      Examples:
+
+          nessie tag -l -> list all tags
+
+          nessie tag -l main -> list only main
+
+          nessie tag -d main -> delete main
+
+          nessie tag -> list all tags
+
+          nessie tag main -> create tag xxx at current head
+
+          nessie tag main test -> create tag xxx at head of test
+
+          nessie tag -f main test -> assign xxx to head of test
+
+    Options:
+      -l, --list     list branches
+      -d, --delete   delete a branches
+      -f, --force    force branch assignment
+      --json         write output in json format.
+      -v, --verbose  Verbose output.
+
+      --help         Show this message and exit.
+
+
+
+Remote Command
+--------------
+Set and view the remote. The ``add`` command is a shortcut to ``nessie config --set endpoint <endpoint>`` and the show
+command functions similarly to the ``git remote show <remote>`` command to show the remote and remote refs.
+
+.. code-block:: bash
+
+    Usage: nessie remote [OPTIONS] COMMAND [ARGS]...
+
+      Set and view remote endpoint.
+
+    Options:
+      --help  Show this message and exit.
+
+    Commands:
+      add   Set current remote.
+      show  Show current remote.
+
+
+Log Command
+-----------
+
+View the commit log. This operats similarly to ``git log`` and shows the log in the terminals pager. Revision range is
+specified as <hash>..<hash> or <hash/ref>.
+
+.. code-block:: bash
+
+    Usage: nessie log [OPTIONS] [REVISION_RANGE] [PATHS]...
```

Comment:
This operation should do the less pagination behavior when in console mode so we can make it paginated later to reduce excessive operations.